src/overlays/actors/ovl_Boss_02/z_boss_02.c

```diff
@@ -97,186 +151,2144 @@ const ActorInit Boss_02_InitVars = {
+                D_809E0420 = 1;
+                goto block_38;
+            }
+            break;
+
+        case 2:
+            if (this->unk_1D14 < 8U) {
+                break;
+            }
+            sp57 = 1;
+            goto block_38;
+
+        case 10:
+            if ((this->unk_1D14 < 30U) && (D_809E0421 != 0) && (globalCtx->state.input[0].press.button & 0xC00F)) {
+                this->unk_1D18++;
+                this->unk_1D78 = 1;
+                this->unk_1D14 = 0;
+                break;
+            }
+
+        label2:
+            if (this->unk_1D14 != 0U) {
+                if (this->unk_1D14 == (u32)(BREG(44) + 10)) {
+                    play_sound(NA_SE_PL_TRANSFORM_NORAML);
+                }
+                Math_ApproachF(&this->unk_1D64, 60.0f, 0.1f, this->unk_1D5C * 640.0f);
+                Math_ApproachF(&this->unk_1D6C, 23.0f, 0.1f, this->unk_1D5C * 150.0f);
+                Math_ApproachF(&this->unk_1D70, 0.01f, 0.1f, 0.003f);
+                Math_ApproachF(&this->unk_1D74, 0.0f, 1.0f, this->unk_1D5C * 100.0f);
+                Math_ApproachF(&this->unk_1D5C, 2.0f, 1.0f, 0.01f);
+            }
+
+            if (this->unk_1D14 == 42U) {
+                this->unk_1D78 = 1;
+            }
+
+            if (this->unk_1D14 > 50U) {
+                D_809E0421 = 1;
+                goto block_38;
+            }
+            break;
+
+        case 11:
+            if (this->unk_1D14 < 8U) {
+                break;
+            }
+
+        block_38:
+        case 20:
+            this->unk_1D18 = 0;
+            func_80169AFC(globalCtx, this->unk_1D22, 0);
+            this->unk_1D22 = 0;
+            func_800EA0EC(globalCtx, &globalCtx->csCtx);
+            this->actor.flags |= 1;
+            player->stateFlags1 &= ~0x100;
+            this->unk_1D70 = 0.01f;
+            func_80165690();
+            break;
+    }
+
+    if (player->transformation == PLAYER_FORM_FIERCE_DEITY) {
+        Actor_SetScale(&player->actor, this->unk_1D70 * 1.5f);
+    } else {
+        Actor_SetScale(&player->actor, this->unk_1D70);
+    }
+
+    globalCtx->envCtx.unk_8C.fogNear = this->unk_1D74;
+
+    if (sp57) {
+        D_809E0422 = 1 - D_809E0422;
+        if (D_809E0422 == 0) {
+            D_809DF5B0 = 1.0f;
+        } else {
+            D_809DF5B0 = 0.1f;
+        }
+
+        this->unk_01AC = D_809DF5B0;
+        if (D_809E0422 == 0) {
+            if (D_809E0434 != NULL) {
+                D_809E0434->unk_203 = 0;
+                D_809E0434->unk_204 = 1.0f;
+                D_809E0434->actor.world.pos.y = 60.0f;
+            }
+
+            player->actor.world.pos.x *= 10.0f;
+            player->actor.world.pos.y -= 3150.0f;
+            player->actor.world.pos.y *= 10.0f;
+            player->unk_B68 = player->actor.world.pos.y;
+            player->actor.world.pos.z *= 10.0f;
+
+            if ((D_809E0434 != NULL) && ((SQ(player->actor.world.pos.z) + SQ(player->actor.world.pos.x)) < 3600.0f)) {
```

Comment:
```suggestion
            if ((D_809E0434 != NULL) && ((SQ(player->actor.world.pos.z) + SQ(player->actor.world.pos.x)) < SQ(60.0f))) {
```

Reply:
```suggestion
            if ((D_809E0434 != NULL) && (SQXZ(player->actor.world.pos) < SQ(60.0f))) {
```
Adding onto this, if the order can be swapped

Reply:
Unfortunately causes regalloc.